main.go

```diff
@@ -110,19 +111,13 @@ func main() {
 	}
 
 	ctx := context.Background()
-	secretConfig, err := teleport.GetConfigFromSecret(ctx, ctrlClient, namespace)
+	config, err := config.GetConfigFromConfigMap(ctx, ctrlClient, namespace)
 	if err != nil {
 		setupLog.Error(err, "unable to get secret config")
 		os.Exit(1)
 	}
 
-	tele := teleport.New(namespace, secretConfig, token.NewGenerator())
-	if tele.TeleportClient, err = teleport.NewClient(ctx, secretConfig.ProxyAddr, secretConfig.IdentityFile); err != nil {
-		setupLog.Error(err, "unable to create teleport client")
-		os.Exit(1)
-	}
-	setupLog.Info("Connected to teleport cluster", "proxyAddr", tele.SecretConfig.ProxyAddr)
-
```

Comment:
Teleport client requires an identity file. However, when the operator starts, the identity may not be available yet, because it's provided by an external resource (tbot).
So, it is not safe to instantiate the client when the operator starts, because it would cause a crash in case the identity is not available.
Therefore, the creation of Teleport client was moved to the reconciliation..